File: web/src/api/channel.ts
import request, { type ListQuery } from './request'

export type ChannelType =
  | 'wecom'
  | 'wechat_kf'
  | 'feishu'
  | 'dingtalk'
  | 'whatsapp'
  | 'telegram'

export interface Channel {
  id: number
  uuid: string
  name: string
  channel_type: ChannelType
  enabled: boolean
  webhook_token: string
  config: Record<string, unknown> | null
  description: string
  created_at: string
  updated_at: string
}

export interface CreateChannelReq {
  name: string
  channel_type: ChannelType
  enabled?: boolean
  webhook_token?: string
  config?: Record<string, unknown>
  description?: string
}

export interface ChannelConversationItem {
  conversation_id: number
  conversation_uuid: string
  title: string
  user_id: string
  sender_id: string
  thread_keys?: string[]
  message_count: number
  last_user_message?: string
  last_reply_message?: string
  updated_at: string
  created_at: string
}

export interface ChannelMessage {
  id: number
  conversation_id: number
  role: string
  content: string
  tool_calls?: unknown
  tool_call_id?: string
  name?: string
  tokens_used: number
  created_at: string
  steps?: unknown[]
}

export const channelApi = {
  list: (params: ListQuery) => request.get('/channels', { params }),
  get: (id: number) => request.get(`/channels/${id}`),
  setEnabled: (id: number, enabled: boolean) => request.patch(`/channels/${id}/enabled`, { enabled }),
  conversations: (id: number, params: ListQuery & { thread_key?: string; sender_id?: string }) =>
    request.get(`/channels/${id}/conversations`, { params }),
  conversationMessages: (channelId: number, conversationId: number, params?: { limit?: number; with_steps?: boolean }) =>
    request.get(`/channels/${channelId}/conversations/${conversationId}/messages`, { params }),
  deleteConversation: (channelId: number, conversationId: number) =>
    request.delete(`/channels/${channelId}/conversations/${conversationId}`),
  create: (data: CreateChannelReq) => request.post('/channels', data),
  update: (id: number, data: Partial<CreateChannelReq> & { config?: Record<string, unknown> | null }) =>
    request.put(`/channels/${id}`, data),
  delete: (id: number) => request.delete(`/channels/${id}`),
}

File: internal/model/channel.go
package model

import "time"

// ChannelType 第三方消息渠道（企微智能机器人走 WebSocket；其余多为 Webhook）。
type ChannelType string

const (
	// ChannelWeCom 企业微信「智能机器人」：WebSocket 长连接，配置 bot_id + secret（实现见 pkg/wecomaibot，协议见官方文档）。
	ChannelWeCom    ChannelType = "wecom"
	ChannelWeChatKF ChannelType = "wechat_kf" // 微信客服
	ChannelFeishu   ChannelType = "feishu"    // 飞书
	ChannelDingTalk ChannelType = "dingtalk"  // 钉钉
	ChannelWhatsApp ChannelType = "whatsapp"  // WhatsApp（如 Meta Cloud API）
	ChannelTelegram ChannelType = "telegram"  // Telegram Bot
)

var ValidChannelTypes = map[ChannelType]struct{}{
	ChannelWeCom:    {},
	ChannelWeChatKF: {},
	ChannelFeishu:   {},
	ChannelDingTalk: {},
	ChannelWhatsApp: {},
	ChannelTelegram: {},
}

func IsValidChannelType(t ChannelType) bool {
	_, ok := ValidChannelTypes[t]
	return ok
}

// Channel 渠道配置：启用后可将下方 Webhook URL 配置到各平台；WebhookToken 用于校验打到本服务的请求。
type Channel struct {
	ID           int64       `json:"id" gorm:"primaryKey;autoIncrement"`
	UUID         string      `json:"uuid" gorm:"uniqueIndex;size:36;not null"`
	Name         string      `json:"name" gorm:"size:200;not null"`
	ChannelType  ChannelType `json:"channel_type" gorm:"size:50;not null;index"`
	Enabled      bool        `json:"enabled" gorm:"not null;default:true"`
	WebhookToken string      `json:"webhook_token,omitzero" gorm:"size:256"` // 调用本服务 Webhook 时携带的共享密钥（可选）
	Config       JSON        `json:"config,omitzero" gorm:"type:text"`       // 平台凭证与参数（JSON）
	Description  string      `json:"description,omitzero" gorm:"type:text"`
	CreatedAt    time.Time   `json:"created_at"`
	UpdatedAt    time.Time   `json:"updated_at"`
}

type CreateChannelReq struct {
	Name         string      `json:"name"`
	ChannelType  ChannelType `json:"channel_type"`
	Enabled      *bool       `json:"enabled,omitzero"`
	WebhookToken string      `json:"webhook_token,omitzero"`
	Config       JSON        `json:"config,omitzero"`
	Description  string      `json:"description,omitzero"`
}

type UpdateChannelReq struct {
	Name         *string `json:"name,omitzero"`
	Enabled      *bool   `json:"enabled,omitzero"`
	WebhookToken *string `json:"webhook_token,omitzero"`
	Config       JSON    `json:"config,omitzero"`
	Description  *string `json:"description,omitzero"`
}

// ChannelConversationItem 渠道会话列表项（用于渠道页面内嵌会话视图）。
type ChannelConversationItem struct {
	ConversationID   int64     `json:"conversation_id"`
	ConversationUUID string    `json:"conversation_uuid"`
	Title            string    `json:"title"`
	UserID           string    `json:"user_id"`
	SenderID         string    `json:"sender_id"`
	ThreadKeys       []string  `json:"thread_keys,omitzero"`
	MessageCount     int64     `json:"message_count"`
	LastUserMessage  string    `json:"last_user_message,omitzero"`
	LastReplyMessage string    `json:"last_reply_message,omitzero"`
	UpdatedAt        time.Time `json:"updated_at"`
	CreatedAt        time.Time `json:"created_at"`
}
